micrometer-spring-legacy/src/samples/java/io/micrometer/spring/samples/MicrometerMetricsPublisherCommand.java

```diff
@@ -0,0 +1,74 @@
+package io.micrometer.spring.samples;
+
+import com.netflix.hystrix.HystrixCircuitBreaker;
+import com.netflix.hystrix.HystrixCommandGroupKey;
+import com.netflix.hystrix.HystrixCommandKey;
+import com.netflix.hystrix.HystrixCommandMetrics;
+import com.netflix.hystrix.HystrixCommandProperties;
+import com.netflix.hystrix.metric.consumer.CumulativeCommandEventCounterStream;
+import com.netflix.hystrix.metric.consumer.RollingCommandEventCounterStream;
+import com.netflix.hystrix.metric.consumer.RollingCommandLatencyDistributionStream;
+import com.netflix.hystrix.metric.consumer.RollingCommandMaxConcurrencyStream;
+import com.netflix.hystrix.metric.consumer.RollingCommandUserLatencyDistributionStream;
+import com.netflix.hystrix.strategy.metrics.HystrixMetricsPublisherCommand;
+import com.netflix.hystrix.util.HystrixRollingNumberEvent;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class MicrometerMetricsPublisherCommand implements HystrixMetricsPublisherCommand {
+    private static final Logger LOG = LoggerFactory.getLogger(MicrometerMetricsPublisherCommand.class);
+
+    private final MeterRegistry meterRegistry;
+    private final HystrixCommandMetrics metrics;
+    private final HystrixCircuitBreaker circuitBreaker;
+    private final List<Tag> tags;
+    private final HystrixCommandProperties properties;
+    private final HystrixCommandKey commandKey;
+
+    public MicrometerMetricsPublisherCommand(MeterRegistry meterRegistry, HystrixCommandKey commandKey, HystrixCommandGroupKey commandGroupKey, HystrixCommandMetrics metrics, HystrixCircuitBreaker circuitBreaker, HystrixCommandProperties properties) {
+        this.meterRegistry = meterRegistry;
+        this.metrics = metrics;
+        this.circuitBreaker = circuitBreaker;
+        this.commandKey = commandKey;
+        this.properties = properties;
+        tags = Tags.zip("commandGroupKey", commandGroupKey.name(), "key", commandKey.name());
+    }
+
+    @Override
+    public void initialize() {
+        Gauge.builder("hystrix.circuit.breaker.open", circuitBreaker, c -> c.isOpen() ? 1 : 0)
+            .tags(tags).register(meterRegistry);
+
+        createCounter("hystrix.circuit.breaker.bad.requests", HystrixRollingNumberEvent.BAD_REQUEST);
```

Comment:
To the extent that `HystrixRollingNumberEvent` states are mutually exclusive, this should be a tag.

Reply:
Are you sure? `HystrixRollingNumberEvent.SHORT_CIRCUITED` and `HystrixRollingNumberEvent.FALLBACK_SUCCESS` seemed to overlap.  

I love the idea of using a tag to group them.

Perhaps I'm seeing the overlap because I crossed groups? https://github.com/Netflix/Hystrix/wiki/Metrics-and-Monitoring#command-execution-event-types-comnetflixhystrixhystrixeventtype

`Command Execution Event Types` would be one metric with various tags, and `Command Fallback Event Types` would be a separate one? That seems like a good fit.